healpy/newvisufunc.py

```diff
@@ -94,41 +111,111 @@ def mollview(
       add graticule
     graticule_labels : bool
       longitude and latitude labels
+    projection_type :  {'aitoff', 'hammer', 'lambert', 'mollweide', 'cart', '3d', 'polar'}
+      type of the plot
+    cb_orientation : {'horizontal', 'vertical'}
+      color bar orientation
+    xlabel : str
+      set x axis label
+    ylabel : str
+      set y axis label
+    longitude_grid_spacing : float
+      set x axis grid spacing
+    latitude_grid_spacing : float
+      set y axis grid spacing
+    override_plot_properties : list
+      Override the following plot proporties: shrink (colorbar size), pad (colorbar padding), 
+      lpad (colorbar label padding), width of the figure, ratio of the height and width of the plot
+    title : str
+      set title of the plot
+    lcolor : str
+      change the color of the longitude tick labels, some color maps make it hard to read black tick labels
     """
+    if not m is None:
+    # auto min and max
+        if min is None:
+            min = m.min()
+        if max is None:
+            max = m.max()
+
+    # do this to find how many decimals are in the colorbar labels, so that the padding in the vertical cbar can done properly
+    def find_number_of_decimals(number):
+        try:
+            return len(str(number).split('.')[1])
+        except:
+            return 0
+    
+    decs = np.max([find_number_of_decimals(min),find_number_of_decimals(max)])
+    if decs >= 3:
+        lpad=-27
+    else:
+        lpad = -9*decs
+
+    # default plot settings
+    r = 0.63
+    if projection_type=="3d":
+        if cb_orientation=='vertical':
+            shrink=0.55;pad=0.02;lpad=lpad;width=11.5
+        if cb_orientation=='horizontal':
+            shrink=0.2;pad=0;lpad=-10;width=14
+    if projection_type == "aitoff" or projection_type ==  "hammer" or  projection_type == "lambert" or projection_type == "mollweide":
+        if cb_orientation=='vertical':
+            shrink=0.6;pad=0.01;lpad=lpad;width=10
+        if cb_orientation=='horizontal':
+            shrink=0.6;pad=0.05;lpad=-8;width=8.5
+    if projection_type =="cart":
+        if cb_orientation=='vertical':
+           shrink=1;pad=0.01;lpad=lpad;width=9.6 ;r=0.42;
+        if  cb_orientation=='horizontal':
+            shrink=0.4;pad=0.1;lpad=-12;width=8.8
+            if xlabel == None:
+                pad=0.01;r=0.63
+    if projection_type=="polar":
+        if  cb_orientation=='vertical':
+            shrink=1;pad=0.01;lpad=lpad;width=10
+        if cb_orientation=='horizontal':
+            shrink=0.4;pad=0.01;lpad=0;width=12
+    if override_plot_properties is not None:
+        print('Overriding default plot properies: shrink= {}, pad={}, lpad={},width={},r={}, to one:  shrink= {}, pad={}, lpad={},width={},r={}'.format(\
```

Comment:
instead of print let's use a warning as in other parts of `healpy` so it is suppressed if one calls `hp.disable_warnings()`

Reply:
Done. When the `plot_properties` are changed there is a warning with old values and then with the updated values. But sometimes only the second warning appears.